yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -262,6 +266,50 @@ private[spark] class ApplicationMaster(
 
   private def runDriver(securityMgr: SecurityManager): Unit = {
     addAmIpFilter()
+    if (isClusterMode) {
+      val containerId = ConverterUtils.toContainerId(
```

Comment:
minor: this code is in `YarnRMClient.getAttemptId`; you could refactor that into a new method instead of copying the code.